### PR TITLE
fix(plugins): rename transformation rules

### PR DESCRIPTION
**Pull request by warren830**
### Summary
1. According to #3468 , we added transformationRules, but because of swagger parsing, we need to add `pluginName` as prefix for all  models.transformationRules

2. change field `refdiff` in GithubTransformationRule as below: 
Refdiff              datatypes.JSONMap `mapstructure:"refdiff,omitempty" json:"refdiff" swaggertype:"object" format:"json"`

### Does this close any open issues?
Relate to #3468 

### Screenshots
image

### Other Information
Any other information that is important to this PR.

### CODE REVIEW

File: plugins/github/api/blueprint_V200_test.go [hunk 1]
```diff
@@ -88,13 +88,17 @@ func TestMakeDataSourcePipelinePlanV200(t *testing.T) {
...
 			},
 			Name:    "github transformation rule",
-			PrType:  "hey,man,wasup",
-			Refdiff: json.RawMessage(`{"tagsPattern": "pattern", "tagsLimit": 10, "tagsOrder": "reverse semver"}`),
+			PrType: "hey,man,wasup",
+			Refdiff: map[string]interface{}{
+				"tagsPattern": "pattern",
+				"tagsLimit":   10,
+				"tagsOrder":   "reverse semver",
+			},
```
Comment (likyh) [inline plugins/github/api/blueprint_V200_test.go:101]: add this change to in PR desc.

File: plugins/jenkins/models/transformation_rule.go [hunk 2]
```diff
@@ -19,12 +19,12 @@ package models
 
 import "github.com/apache/incubator-devlake/models/common"
 
-type TransformationRules struct {
+type JenkinsTransformationRule struct {
 	common.Model      `mapstructure:"-"`
-	Name              string `gorm:"type:varchar(255)" mapstructure:"name" json:"name"`
+	Name              string `gorm:"type:varchar(255)"`
```
Comment (likyh) [inline plugins/jenkins/models/transformation_rule.go:24]: why delete `mapstructure`?

File: plugins/jira/api/transformation_rule.go [hunk 3]
```diff
@@ -86,20 +87,32 @@ func UpdateTransformationRule(input *core.ApiResourceInput) (*core.ApiResourceOu
...
+	blob, err := json.Marshal(rule.TypeMappings)
+	if err != nil {
+		return nil, errors.Default.Wrap(err, "error marshaling TypeMappings")
+	}
+	return &models.JiraTransformationRule{
+		EpicKeyField:               rule.EpicKeyField,
+		StoryPointField:            rule.StoryPointField,
+		RemotelinkCommitShaPattern: rule.RemotelinkCommitShaPattern,
+		TypeMappings:               blob,
+	}, nil
```
Comment (likyh) [inline plugins/jira/api/transformation_rule.go:107]: this code delete in https://github.com/apache/incubator-devlake/pull/3827. It looks like a rebase bug.

File: plugins/jenkins/models/transformation_rule.go (hunk 2 above)
Comment (warren830) [reply plugins/jenkins/models/transformation_rule.go]: fixed

File: plugins/jira/api/transformation_rule.go (hunk 3 above)
Comment (warren830) [reply plugins/jira/api/transformation_rule.go]: fixed

**Review by likyh — APPROVED** [review]
LGTM if ci passed

**Review by likyh — APPROVED** [review]

**Pull request merged**